### ISSUE TITLE
Bug: [no-unnecessary-condition] does not report unnecessary string comparisons

### ISSUE THREAD

**Issue by FloEdelmann** [typescript-eslint/typescript-eslint#9482]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.3.3&fileType=.tsx&code=CYUwxgNghgTiAEYD2A7AzgF3gMyUgXPJjAJYoDmAUCdvABS5ICU8A3vAL6UgRoI30AhIxasuPPvAEM8ozpWq06w2W3kT%2BSlczXjem%2BiN0LQkWAmTosAI1iFreCCCgpF9WzFF7J0wR7neBnT%2Bxm7KIeyBUloR6vrR7rABlEA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tiacTJTIAhtEK0yHJgBNK%2BSpPRRE0aB2iRwYAL4gtQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
declare const foo: string
if (foo) { }
else if (!foo) {}
else if (foo) { }

if (!foo) { }
else if (!foo) { }
else if (foo) { }

declare const bar: boolean
if (bar) {}
else if (!bar) { }
else if (bar) { }

if (!bar) { }
else if (!bar) { }
else if (bar) { }
```

### ESLint Config

```javascript
{
  "rules": {
    "@typescript-eslint/no-unnecessary-condition": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

All `else if` conditions should be reported, because they are all unnecessary.

### Actual Result

The `else if` conditions are reported correctly for the boolean variable, but not for the string variable.
